internal/pkg/policy/policy_output.go

```diff
@@ -475,8 +475,9 @@ if (ctx._source['outputs']['%s']==null)
 			source.WriteString(fmt.Sprintf(`
 if (ctx._source['outputs']['%s'].%s==null)
   {ctx._source['outputs']['%s'].%s=new ArrayList();}
-ctx._source['outputs']['%s'].%s.add(params.%s);
-`, outputName, field, outputName, field, outputName, field, field))
+if (!ctx._source['outputs']['%s'].%s.contains(params.%s))  
```

Comment:
only adding `toRetireAPIKey` if does not exist, because I noticed the same API key was added multiple times.
the comment in L473 says `It's an array that gets deleted when the keys are invalidated.` though I haven't found where the invalidated api keys are deleted from the retire list, does anyone know?

Reply:
I think it used to happen when it was the `DefaultAPIKeyHistory` list. Now I think we keep the history. I was trying to find when it'd get deleted and I also did not find. The comment was probably left there after the behaviour changed.

Reply:
this might be a bug if we don't ever clear the list of retired api keys

Reply:
I don't remember right now, but I think the idea was to keep the history. Besides with the possibility to update API keys permissions the number of retired API keys would be rather small